src/output.cpp

```diff
@@ -0,0 +1,736 @@
+      "<http://openmc.readthedocs.io/en/latest/license.html>\n";
+  }
+}
+
+//==============================================================================
+
+void print_columns()
+{
+  if (settings::entropy_on) {
+    std::cout <<
+      "  Bat./Gen.      k       Entropy         Average k \n"
+      "  =========   ========   ========   ====================\n";
+  } else {
+    std::cout <<
+      "  Bat./Gen.      k            Average k\n"
+      "  =========   ========   ====================\n";
+  }
+}
+
+//==============================================================================
+
+void print_generation()
+{
+  // Save state of cout
+  auto f {std::cout.flags()};
+
+  // Determine overall generation and number of active generations
+  int i = overall_generation() - 1;
+  int n = simulation::current_batch > settings::n_inactive ?
+    settings::gen_per_batch*simulation::n_realizations + simulation::current_gen : 0;
+
+  // Set format for values
+  std::cout << std::fixed << std::setprecision(5);
+
+  // write out information batch and option independent output
+  std::cout << "  "  << std::setw(9) <<  std::to_string(simulation::current_batch)
+    + "/" + std::to_string(simulation::current_gen) << "   " << std::setw(8)
+    << simulation::k_generation[i];
+
+  // write out entropy info
+  if (settings::entropy_on) {
+    std::cout << "   " << std::setw(8) << simulation::entropy[i];
+  }
+
+  if (n > 1) {
+    std::cout << "   " << std::setw(8) << simulation::keff << " +/-"
+      << std::setw(8) << simulation::keff_std;
+  }
+  std::cout << '\n';
+
+  // Restore state of cout
+  std::cout.flags(f);
+}
+
+//==============================================================================
+
+void print_batch_keff()
+{
+  // Save state of cout
+  auto f {std::cout.flags()};
+
+  // Determine overall generation and number of active generations
+  int i = simulation::current_batch*settings::gen_per_batch - 1;
+  int n = simulation::n_realizations*settings::gen_per_batch;
+
+  // Set format for values
+  std::cout << std::fixed << std::setprecision(5);
+
+  // write out information batch and option independent output
+  std::cout << "  "  << std::setw(9) <<  std::to_string(simulation::current_batch)
+    + "/" + std::to_string(settings::gen_per_batch) << "   " << std::setw(8)
+    << simulation::k_generation[i];
+
+  // write out entropy info
+  if (settings::entropy_on) {
+    std::cout << "   " << std::setw(8) << simulation::entropy[i];
+  }
+
+  if (n > 1) {
+    std::cout << "   " << std::setw(8) << simulation::keff << " +/-"
+      << std::setw(8) << simulation::keff_std;
+  }
+  std::cout << std::endl;
+
+  // Restore state of cout
+  std::cout.flags(f);
+}
+
+//==============================================================================
+
+void show_time(const char* label, double secs, int indent_level=0)
+{
+  std::cout << std::string(2*indent_level, ' ');
+  int width = 33 - indent_level*2;
```

Comment:
Question: Is there a significance of 33 here (is it the sum of a number of other widths)?  Same for line 447.  If so I would make it a class-level constant or otherwise tie it to a more central config.